operator/pkg/ciliumidentity/identity.go

```diff
@@ -3,8 +3,109 @@
 
 package ciliumidentity
 
-import "github.com/cilium/cilium/pkg/k8s/resource"
+import (
+	"context"
+	"time"
+
+	"k8s.io/client-go/util/workqueue"
+
+	ciliumapiv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
 
 func cidResourceKey(cidName string) resource.Key {
 	return resource.Key{Name: cidName}
 }
+
+func (c *Controller) processCiliumIdentityEvents(ctx context.Context) error {
+	for event := range c.ciliumIdentity.Events(ctx) {
+		switch event.Kind {
+		case resource.Upsert:
+			c.logger.Debug("Got Upsert CID event",
+				logfields.CIDName, event.Key.String())
+			c.onCiliumIdentityUpsertEvent(event.Object)
+		case resource.Delete:
+			c.logger.Debug("Got Delete CID event",
+				logfields.CIDName, event.Key.String())
+			c.onCiliumIdentityDeleteEvent(event.Object)
+		}
+		event.Done(nil)
+	}
+	return nil
+}
+
+func (c *Controller) onCiliumIdentityUpsertEvent(cid *ciliumapiv2.CiliumIdentity) {
+	c.enqueueCIDReconciliation(cidResourceKey(cid.Name), 0)
+}
+
+func (c *Controller) onCiliumIdentityDeleteEvent(cid *ciliumapiv2.CiliumIdentity) {
+	c.enqueueCIDReconciliation(cidResourceKey(cid.Name), 0)
+}
+
+func (c *Controller) initCIDQueue() {
+	c.logger.Info("CID controller work queue configuration for CID",
+		logfields.WorkQueueSyncBackOff, defaultSyncBackOff,
+		logfields.WorkQueueMaxSyncBackOff, maxSyncBackOff)
+
+	c.cidQueue = workqueue.NewRateLimitingQueueWithConfig(
+		workqueue.NewItemExponentialFailureRateLimiter(defaultSyncBackOff, maxSyncBackOff),
+		workqueue.RateLimitingQueueConfig{Name: "ciliumidentity_identity"})
+}
+
+// runCIDWorker runs a worker thread that just dequeues items, processes them, and
+// marks them done. You may run as many of these in parallel as you wish; the
+// work queue guarantees that they will not end up processing the same CID
+// at the same time
+func (c *Controller) runCIDWorker(_ context.Context) error {
+	c.logger.Info("Starting CID worker in CID controller")
```

Comment:
I think we should additionally watch the passed in context  vs the controller context as this would indicate processing on the queue is being terminated?

Reply:
added a case for `case <-context.Done():`